week_4/exceptions_in_the_class_rational/exceptions_in_the_class_rational.cpp

```diff
@@ -116,39 +116,40 @@ class Rational {
     return result;
   }
 
- private:
-  int p;
-  int q;
-};
+  friend std::istream& operator>>(std::istream& stream, Rational& r) {
```

Comment:
@Ypostolov Why have you made this is a friend function, but not class member-method?

Reply:
Fixed in 6837310212b92d0b5ad8047ce10bcd625a6c1f84
Please take a look.